api/elections.go

```diff
@@ -281,66 +281,104 @@ func (a *API) electionScrutinyHandler(msg *apirest.APIdata, ctx *httprouter.HTTP
 	}
-	// check process results are the same
-	var results [][]*big.Int
-	if len(process.Results) > 0 {
-		firstResult := process.Results[0]
-		for _, processResult := range process.Results[1:] {
-			if processResult == nil {
-				log.Errorf("invalid process result")
+	if process.Results == nil {
+		return fmt.Errorf("no results available")
+	}
+
+	// intermediateResults is an array of the results that are signed
+	// by oracles. There can be void results see vochain/state/process.go#L330
+	// but we want to ignore them. The way we can know if a result is void is
+	// by checking if the oracle address is empty.
+	intermediateResults := []*models.ProcessResult{}
+	for k, processResult := range process.Results {
+		if processResult == nil {
+			log.Warnw("nil process result",
+				map[string]interface{}{
+					"electionID":    fmt.Sprintf("%x", electionID),
+					"results index": k,
+				})
+			continue
+		}
+		if len(processResult.OracleAddress) == 0 { // check oracle sent the results otherwise ignore
+			continue
+		}
+		intermediateResults = append(intermediateResults, processResult)
+	} // now we have an array of non void results set by a set of oracles via SetProcessResults
+
+	if len(intermediateResults) == 0 {
+		return fmt.Errorf("no results available")
+	}
+
+	// Results are being compared because we want to make sure that
+	// all oracles are sending the same results.
+	equalResults := func(baseResult, otherResult *models.ProcessResult) bool {
+		// if the number of votes is different, the results are different
+		if len(baseResult.Votes) != len(otherResult.Votes) {
+			log.Warnw("different number of votes",
+				map[string]interface{}{
+					"baseResult number of votes":  len(baseResult.Votes),
+					"otherResult number of votes": len(otherResult.Votes),
+				})
+			return false
+		}
+		for k, vote1 := range otherResult.Votes {
+			vote2 := baseResult.Votes[k]
+			if vote1 == nil {
+				log.Warnw("invalid question result (nil)",
+					map[string]interface{}{
+						"question result index": k,
+					})
 				continue
 			}
-			if len(firstResult.Votes) != len(processResult.Votes) {
-				log.Error("votes length do not match")
+			if len(vote1.Question) != len(vote2.Question) {
+				log.Warnw("question length do not match",
+					map[string]interface{}{
+						"baseResult question length":  len(vote2.Question),
+						"otherResult question length": len(vote1.Question),
+					})
 				continue
 			}
-			// if consensus results do not match, return error
-			for k, questionResult := range processResult.Votes {
-				if questionResult == nil {
-					log.Errorf("invalid question result")
-					continue
-				}
-				if len(questionResult.Question) != len(firstResult.Votes[k].Question) {
-					log.Errorf("question length do not match")
+			for kk, question1 := range vote1.Question {
+				question2 := vote2.Question[kk]
+				if question1 == nil {
+					log.Warnw("invalid question option (nil)",
```

Comment:
Please rebase with master. The syntax for using log key/value functions has changed, now it expects a list of key/values instead of a map (this way is more standard and comfortable). The correct way to use this functions would be something like: `log.Warnw("invalid question", "question", nil, "index", kk)`